<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QTminiCcompilerClass</class>
 <widget class="QMainWindow" name="QTminiCcompilerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>673</width>
    <height>630</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>540</width>
    <height>630</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>QTminiCcompiler</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="minimumSize">
    <size>
     <width>540</width>
     <height>600</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="minimumSize">
         <size>
          <width>35</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>代码文件</string>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="CodePathEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="openCodeBtn">
        <property name="text">
         <string>打开代码</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>文法</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="GrammarPathEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="openGrammarBtn">
        <property name="text">
         <string>打开文法</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0" colspan="2">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="lexicalAnaBtn">
        <property name="text">
         <string>词法分析</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="syntacticAnaBtn">
        <property name="text">
         <string>语法分析</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="semanticAnaBtn">
        <property name="text">
         <string>语义分析</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="genCodeBtn">
        <property name="text">
         <string>生成代码</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>源代码</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="CodeText"/>
      </item>
     </layout>
    </item>
    <item row="2" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>文法</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEdit"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>673</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="QTminiCcompiler.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>QTminiCcompilerClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openCodeBtn</sender>
   <signal>clicked()</signal>
   <receiver>QTminiCcompilerClass</receiver>
   <slot>openCodeFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGrammarBtn</sender>
   <signal>clicked()</signal>
   <receiver>QTminiCcompilerClass</receiver>
   <slot>openGrammarFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>625</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lexicalAnaBtn</sender>
   <signal>clicked()</signal>
   <receiver>QTminiCcompilerClass</receiver>
   <slot>LexicalScan()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>syntacticAnaBtn</sender>
   <signal>clicked()</signal>
   <receiver>QTminiCcompilerClass</receiver>
   <slot>syntacticAna()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>336</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openCodeFile()</slot>
  <slot>openGrammarFile()</slot>
  <slot>LexicalScan()</slot>
  <slot>syntacticAna()</slot>
 </slots>
</ui>
